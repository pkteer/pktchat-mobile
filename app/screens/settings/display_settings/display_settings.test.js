// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {DeviceTypes} from '@constants';
import Preferences from '@mm-redux/constants/preferences';
import SettingsItem from '@screens/settings/settings_item';
import {shallowWithIntl} from '@test/intl-test-helper';

import DisplaySettings from './display_settings';

describe('DisplaySettings', () => {
    const baseProps = {
        theme: Preferences.THEMES.denim,
        enableTheme: false,
        enableTimezone: false,
        componentId: 'component-id',
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <DisplaySettings {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(SettingsItem).length).toBe(1);
        wrapper.setProps({enableTheme: true});
        expect(wrapper.find(SettingsItem).length).toBe(2);
        wrapper.setProps({enableTimezone: true});
        expect(wrapper.find(SettingsItem).length).toBe(3);
    });

    test('should match snapshot on Tablet devices', () => {
        DeviceTypes.IS_TABLET = true;

        const wrapper = shallowWithIntl(
            <DisplaySettings {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(SettingsItem).length).toBe(2);
        wrapper.setProps({enableTheme: true});
        expect(wrapper.find(SettingsItem).length).toBe(3);
        wrapper.setProps({enableTimezone: true});
        expect(wrapper.find(SettingsItem).length).toBe(4);
    });
});
