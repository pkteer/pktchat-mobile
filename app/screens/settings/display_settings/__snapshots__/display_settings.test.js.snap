// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplaySettings should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="display_settings.screen"
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "backgroundColor": "rgba(63,67,80,0.06)",
        "flex": 1,
        "paddingTop": 35,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(63,67,80,0.1)",
          "height": 1,
        }
      }
    />
    <SettingsItem
      defaultMessage="Clock Display"
      i18nId="mobile.advanced_settings.clockDisplay"
      iconName="clock-outline"
      isDestructor={false}
      isLink={false}
      onPress={[Function]}
      separator={false}
      showArrow={true}
      testID="display_settings.clock_display.action"
      theme={
        Object {
          "awayIndicator": "#ffbc1f",
          "buttonBg": "#1c58d9",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3f4350",
          "codeTheme": "github",
          "dndIndicator": "#d24b4e",
          "errorTextColor": "#d24b4e",
          "linkColor": "#386fe5",
          "mentionBg": "#ffffff",
          "mentionColor": "#1e325c",
          "mentionHighlightBg": "#ffd470",
          "mentionHighlightLink": "#1b1d22",
          "newMessageSeparator": "#cc8f00",
          "onlineIndicator": "#3db887",
          "sidebarBg": "#1e325c",
          "sidebarHeaderBg": "#192a4d",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarTeamBarBg": "#14213e",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#5d89ea",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#28427b",
          "sidebarUnreadText": "#ffffff",
          "type": "Denim",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(63,67,80,0.1)",
          "height": 1,
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;

exports[`DisplaySettings should match snapshot on Tablet devices 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="display_settings.screen"
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "backgroundColor": "rgba(63,67,80,0.06)",
        "flex": 1,
        "paddingTop": 35,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(63,67,80,0.1)",
          "height": 1,
        }
      }
    />
    <SettingsItem
      defaultMessage="Sidebar"
      i18nId="mobile.display_settings.sidebar"
      iconName="dock-left"
      isDestructor={false}
      isLink={false}
      onPress={[Function]}
      separator={true}
      showArrow={true}
      theme={
        Object {
          "awayIndicator": "#ffbc1f",
          "buttonBg": "#1c58d9",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3f4350",
          "codeTheme": "github",
          "dndIndicator": "#d24b4e",
          "errorTextColor": "#d24b4e",
          "linkColor": "#386fe5",
          "mentionBg": "#ffffff",
          "mentionColor": "#1e325c",
          "mentionHighlightBg": "#ffd470",
          "mentionHighlightLink": "#1b1d22",
          "newMessageSeparator": "#cc8f00",
          "onlineIndicator": "#3db887",
          "sidebarBg": "#1e325c",
          "sidebarHeaderBg": "#192a4d",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarTeamBarBg": "#14213e",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#5d89ea",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#28427b",
          "sidebarUnreadText": "#ffffff",
          "type": "Denim",
        }
      }
    />
    <SettingsItem
      defaultMessage="Clock Display"
      i18nId="mobile.advanced_settings.clockDisplay"
      iconName="clock-outline"
      isDestructor={false}
      isLink={false}
      onPress={[Function]}
      separator={false}
      showArrow={true}
      testID="display_settings.clock_display.action"
      theme={
        Object {
          "awayIndicator": "#ffbc1f",
          "buttonBg": "#1c58d9",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3f4350",
          "codeTheme": "github",
          "dndIndicator": "#d24b4e",
          "errorTextColor": "#d24b4e",
          "linkColor": "#386fe5",
          "mentionBg": "#ffffff",
          "mentionColor": "#1e325c",
          "mentionHighlightBg": "#ffd470",
          "mentionHighlightLink": "#1b1d22",
          "newMessageSeparator": "#cc8f00",
          "onlineIndicator": "#3db887",
          "sidebarBg": "#1e325c",
          "sidebarHeaderBg": "#192a4d",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarTeamBarBg": "#14213e",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#5d89ea",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#28427b",
          "sidebarUnreadText": "#ffffff",
          "type": "Denim",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(63,67,80,0.1)",
          "height": 1,
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;
